frontend/js/main.js

```diff
@@ -0,0 +1,2203 @@
+      }
+    })
+    tableCardHeading.appendChild(editSpannerSpan);
+
+    tableCardCollapse = document.createElement("div");
+    tableCardCollapse.setAttribute("id", Object.keys(schemaConversionObj.SrcSchema)[i]);
+    tableCardCollapse.className = "collapse reportCollapse";
+
+    tableCardContent = document.createElement("div");
+    tableCardContent.className = 'mdc-card mdc-card-content table-card-border' + mdcCardBorder(conversionRateResp[src_table_name[i]]);
+
+    tableAccContent = document.createElement("div");
+    tableAccContent.className = "acc-card-content";
+
+    // creating column headers for each table
+    tableColHeaders = [];
+    for (var m = 0; m < 6; m++) {
+      if (m % 2 == 0) {
+        tableColHeaders.push(sourceTableFlag);
+      }
+      else
+        tableColHeaders.push('Spanner');
+    }
+
+    // appending column headers to the table
+    table = document.createElement("table");
+    table_header2 = document.createElement('thead');
+    table.appendChild(table_header2);
+    table.setAttribute('id', 'src-sp-table' + i);
+    table.className = 'acc-table';
+
+    tr = table_header2.insertRow(-1);
+    th1 = document.createElement('th');
+    th1.innerHTML = 'Column Name';
+    th1.className = 'acc-column';
+    th1.setAttribute('colspan', 2);
+    tr.appendChild(th1);
+
+    th2 = document.createElement('th');
+    th2.innerHTML = 'Data Type';
+    th2.className = 'acc-column';
+    th2.setAttribute('colspan', 2);
+    tr.appendChild(th2);
+
+    th3 = document.createElement('th');
+    th3.innerHTML = 'Constraints';
+    th3.className = 'acc-column';
+    th3.setAttribute('colspan', 2);
+    tr.appendChild(th3);
+
+    tr = table_header2.insertRow(-1);
+    for (var j = 0; j < tableColHeaders.length; j++) {
+      th = document.createElement("th");
+      if (j % 2 == 0) {
+        if (j == 0) {
+          th.className = "acc-table-th-src src-tab-cell";
+        }
+        else {
+          th.className = "acc-table-th-src";
+        }
+      }
+      else {
+        th.className = "acc-table-th-spn";
+      }
+      th.innerHTML = tableColHeaders[j];
+      tr.appendChild(th);
+    }
+
+    table_body = document.createElement('tbody');
+    table.appendChild(table_body);
+    columnsLength = Object.keys(schemaConversionObj.ToSpanner[sp_table.Name].Cols).length;
+
+    spannerColumnsIterator(i);
+
+    tableAccContent.appendChild(table)
+    tableCardContent.appendChild(tableAccContent)
+    tableCardCollapse.appendChild(tableCardContent)
+    tableCard.appendChild(tableCardCollapse)
+    li.appendChild(tableCard);
+
+    if (sp_table.Fks != null) {
+      foreignKeyHandler(i, sp_table.Fks);
+    }
+    interleaveHandler(i, interleaveApiCallResp[i]);
+    if (JSON.parse(localStorage.getItem('summaryReportContent')) != undefined) {
+      createSummaryForEachTable(i, JSON.parse(localStorage.getItem('summaryReportContent')));
+    }
+  }
+  accordion.appendChild(reportUl);
+  z--;
+  while (z != -1) {
```

Comment:
1. This while loop is required to count the number of constraints available in source db cell. It will execute once all the tables are created. Condition used inside while loop was not proper, now it is changed to while(z >= 0)
2. z is declared now
3. Also now the variable name is changed to constraintCount from z